primitive-types/impls/serde/src/serialize.rs

```diff
@@ -30,51 +54,38 @@ pub fn serialize_uint<S>(bytes: &[u8], serializer: S) -> Result<S::Ok, S::Error>
 		return serializer.serialize_str("0x0");
 	}
 
-	let hex: String = ::rustc_hex::ToHex::to_hex(bytes);
-	let has_leading_zero = !hex.is_empty() && &hex[0..1] == "0";
-	serializer.serialize_str(
-		&format!("0x{}", if has_leading_zero { &hex[1..] } else { &hex })
-	)
+	serializer.serialize_str(to_hex(slice, bytes, true))
```

Comment:
I don't want to re-review the changes here, but I do wonder why we're re-writing hex-ing code. Isn't `rustc_hex` doing a good job?

Reply:
I think the old code used `rustc_hex-1.0`; switching to `2.0` might have been enough to remove the heap allocations.

Reply:
Correct me if I'm wrong, but it seems like `rustc-hex 2.0` [uses](https://github.com/debris/rustc-hex/blob/3e773fee3992d8a2ecfcc22cc99141c2a6bf76c9/src/lib.rs#L182) `FromIterator<u8>` as a generic return type, but this trait is [only implemented](https://doc.rust-lang.org/std/iter/trait.FromIterator.html#implementors) for boxed types.